common/priorityQueue.py

```diff
@@ -0,0 +1,81 @@
+import bisect
+
+class PriorityQueue:
+    def __init__(self):
+        self._queue = []
+        self._size = 0
+        self._id = 0
```

Comment:
The self._id is used to sort items in the priority queue. Its main purpose is to deal with items that are the exact same so  insort_left can put them in the right place.